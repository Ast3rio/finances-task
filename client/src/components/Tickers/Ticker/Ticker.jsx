import React, {useEffect, useMemo, useState} from 'react';
import styles from "./Ticker.module.scss";
import {getDate, getTime} from "../../../utils/getDate";
import {arrowIco} from "../../../assets/imagesConstants";

const different = (percent, newPercent) => {
    let result = Math.round((newPercent - percent)*100)/100;
    return result < 0 ? 1 : 0;
}

export const Ticker = ({ticker, price, change_percent, last_trade_time}) => {

    const [newPrice, setNewPrice] = useState(0)
    const [active, setActive] = useState(0)

    const percentDifferent = useMemo(() => 
        different(price, newPrice), 
        [price, newPrice]
    )

    useEffect(() => {
        setNewPrice(price)
        setActive(percentDifferent)
    }, [price])

    return (
        <li className={styles.item}>
            <div className={styles.ticker}>
                {ticker}
            </div>
            <div className={styles.price}>
                {`$ ${price}`}
            </div>
            <div className={styles.date}>
                {`${getDate(last_trade_time)} ${getTime(last_trade_time)}`}
            </div>
            <div className={`${styles.percent} ${active === 1 && styles.active}`}>
                <img className={styles.percent_img} src={arrowIco} alt="arrow"/>
                {`${change_percent} %`}
            </div>
        </li>
    );
};
