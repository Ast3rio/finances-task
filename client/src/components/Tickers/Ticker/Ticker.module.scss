.item{
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 40px;
  padding: 0 20px;
  border: 1px solid #efefef;
  border-left: none;
  border-right: none;
  transition: all .3s;
  &:hover{
    background-color: #efefef;
  }
}

.ticker{
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #211f1f;
  border-radius: 10px;
  color: #fff;
  text-transform: uppercase;
  width: 80px;
  height: 30px;
  margin-right: 20px;
  font-size: 1.4rem;
}

.price{
  width: 80px;
  margin-right: 20px;
}

.date{
  display: flex;
  align-items: center;
  justify-content: center;
  width: 200px;
  margin-right: 20px;
}

.percent{
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #e5b8b8;
  border-radius: 10px;
  color: #fff;
  text-transform: uppercase;
  width: 80px;
  height: 30px;
  font-size: 1.4rem;
  transition: all .5s;
  &.active{
    background-color: #8acd73;
    .percent_img{
      transform: rotate(-90deg);
    }
  }
  &_img{
    width: 23px;
    height: 18px;
    transform: rotate(90deg);
    transition: all .5s;
    filter: invert(100%) sepia(0%) saturate(7500%) hue-rotate(181deg) brightness(106%) contrast(100%);
    img{
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }
}