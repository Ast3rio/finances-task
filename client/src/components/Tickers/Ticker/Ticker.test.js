import React from "react";
import {shallow} from "enzyme";
import {Ticker} from "./Ticker";

describe('Test <Ticker/>', () => {
    const ticker = shallow(<Ticker />);

    it('Ticker have rendered correctly', () => {
        expect(ticker).toMatchSnapshot();
    })

    it('Ticker is empty object', () => {
        expect(ticker.props.ticker).toBeString();
    })
})

