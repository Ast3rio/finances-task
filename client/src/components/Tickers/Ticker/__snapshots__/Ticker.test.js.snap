// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Ticker/> Ticker have rendered correctly 1`] = `ShallowWrapper {}`;
